lib/posix-poll/epoll.c

```diff
@@ -562,8 +562,8 @@ int uk_sys_epoll_pwait2(const struct uk_file *epf, struct epoll_event *events,
 		}
 		uk_file_runlock(epf);
 
-		/* If lvlev, update pollin back in */
-		if (lvlev)
+		/* If lvlev or limited by maxevents, update pollin back in */
+		if (lvlev || nout == maxevents)
 			uk_file_event_set(epf, UKFD_POLLIN);
```

Comment:
Doesn't this function return something? Is there a reasion we are not checking for return values?

Reply:
In fact I don't see its return values being checked at any of its callsite :thinking:

Reply:
`uk_file_event_set` returns the previous events, as a convenience due to the fact that event setting is done atomically so there's no other way to get that previous value.

In this case we don't care what the previous events were. I used to cast such invocations to void, but this turned out to be _everywhere_, visually polluting the code, not to mention increasing the pressure on keeping everything withing 80 cols with size 8 tabs :stuck_out_tongue: ; so I stopped that.

Reply:
Oh well I guess we also call `printf` while ignoring return value...